### PR TITLE
Refactor + Extreme toggle

### PR DESCRIPTION
**Pull request by Seirade**

### CODE REVIEW

File: common.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	foreach($tags as $tag)
+	{
+		if(in_array($tag, $nsfw))
+		{
+			return true;
+		}
+	}
+	
+	return false;
+}
```
Comment (puigru) [inline common.php]: Newline at EOF

File: common.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+foreach($json['tagFilters'] as $filter)
+{
+	if($filter['extreme'])
+	{
+		array_push($nsfw, ... $filter['tags']);
+	}
+}
+
+function is_extreme($game, $nsfw)
```
Comment (puigru) [inline common.php]: isExtreme

File: css/style.css [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+	object-fit: contain;
+	width: 250px;
+	max-height: 250px;
+}
+
+.blur
+{
+	filter: blur(8px);
+	cursor: pointer;
+}
```
Comment (puigru) [inline css/style.css]: Newline at EOF

File: index.php [hunk 4]
```diff
@@ -1,105 +1,68 @@
...
+						</form>
+					</div>
+					<div class="modal-footer">
+						<button type="button" class="btn btn-primary" data-dismiss="modal">Close</button>
+					</div>
+				</div>
+			</div>
+		</div>
+	</body>
+</html>
```
Comment (puigru) [inline index.php]: Newline at EOF

File: js/search.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+	$('#settings-dialog').on('hidden.bs.modal', function() {
+		settings = $('#settings').form();
+		localStorage.setItem('settings', JSON.stringify(settings));
+		refresh();
+	});
+	$('#search').on('submit', function(e) {
+		e.preventDefault();
+	});
+	$('#search .form-control').on('input change', refresh);
+}
```
Comment (puigru) [inline js/search.js]: Newline at EOF

File: index.php [hunk 6]
```diff
@@ -1,105 +1,68 @@
 <!DOCTYPE html>
 <html>
-<meta charset="utf-8">
-<title>Flashpoint Browser</title>
-<script src="https://code.jquery.com/jquery-3.5.1.min.js" crossorigin="anonymous"></script>
-<script src="https://cdn.jsdelivr.net/npm/popper.js@1.16.0/dist/umd/popper.min.js" integrity="sha384-Q6E9RHvbIyZFJoft+2mJbHaEWldlvI9IOYy5n3zV9zzTtmI3UksdQRVvoxMfooAo" crossorigin="anonymous"></script>
-<script src="https://stackpath.bootstrapcdn.com/bootstrap/4.5.0/js/bootstrap.min.js" integrity="sha384-OgVRvuATP1z7JjHLkuOU7Xw704+h835Lr+6QL9UvYjZE3Ipu6Tp75j7Bh/kR0JKI" crossorigin="anonymous"></script>
-<link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css" integrity="sha384-Vkoo8x4CGsO3+Hhxv8T/Q5PaXtkKtu6ug5TOeNV6gBiFeWPGFN9MuhOf23Q9Ifjh" crossorigin="anonymous">
-<link rel="stylesheet" href="https://maxst.icons8.com/vue-static/landings/line-awesome/line-awesome/1.3.0/css/line-awesome.min.css">
```
Comment (puigru) [inline index.php:9]: Why did you remove line-awesome?

File: index.php [hunk 7]
```diff
@@ -1,105 +1,68 @@
...
-    e.preventDefault();
-});
-$('#search .form-control').on('input change', refresh);
-</script>
+	<head>
+		<meta charset="utf-8">
+		
+		<title>Flashpoint Browser</title>
+		
+		<script src="js/jquery-3.5.1.min.js"></script>
```
Comment (puigru) [inline index.php]: Not convinced by the whole pulling the dependencies into the source tree, but ok.

File: search.php [hunk 8]
```diff
@@ -1,43 +1,46 @@
 <?php
+include 'common.php';
```
Comment (puigru) [inline search.php]: require

File: search.php [hunk 9]
```diff
@@ -1,43 +1,46 @@
...
-<?php endfor ?>
+
+for($i = 0; $game = $result->fetchArray(SQLITE3_ASSOC); ++$i)
+{
+	if(!$extreme && is_extreme($game, $nsfw))
+	{
+		continue;
+	}
+	
+	echo '<div class="game">';
```
Comment (puigru) [inline search.php]: Please use PHP's alternative syntax for control structures when templating HTML.

File: search.php [hunk 10]
```diff
@@ -1,43 +1,46 @@
...
+	echo '<a href="#" data-toggle="collapse" data-target="#game-' . $i . '">[' . $game['platform'] . "] " . $game['title'] . '</a>';
+	echo is_extreme($game, $nsfw) ? ' 🔞' : '';
```
Comment (puigru) [inline search.php]: I suggest `<i class="las la-radiation"></i>`

File: search.php [hunk 11]
```diff
@@ -1,43 +1,46 @@
...
+	echo empty($game['activeDataId']) ? '<span title="This game is in LEGACY format."> 📜</span>' : '';
+	echo isset($_POST['fpurl']) ? '<a href="flashpoint://' . $game['id'] . '"> ↪️</a>' : '';
+	echo '<div class="game-details collapse" id="game-' . $i . '" data-id="' . $game['id'] . '">Loading...</div>';
+}
```
Comment (puigru) [inline search.php]: Newline at EOF

File: view.php [hunk 12]
```diff
@@ -1,39 +1,50 @@
...
 $stmt = $db->prepare('SELECT * FROM game WHERE id = :id');
 $stmt->bindValue(':id', $id);
 $result = $stmt->execute();
 $game = $result->fetchArray(SQLITE3_ASSOC);
-if (!$game) {
-    header('HTTP/1.1 404 Not Found');
-    die('Game not found');
+$blur = is_extreme($game, $nsfw);
+
+if(!$game)
```
Comment (puigru) [inline view.php]: Please keep the brace style

File: view.php [hunk 13]
```diff
@@ -1,39 +1,50 @@
...
+<img class="thumb<?php echo $blur ? ' blur' : '' ?>" src="<?php echo "$baseurl/Screenshots/$img"; ?>" />
+
 <pre>
 <?php
-foreach($keys as $key => $name) {
-    if ($game[$key]) {
-        echo "$name: {$game[$key]}\n";
-    }
-}
+	foreach($keys as $key => $name)
```
Comment (puigru) [inline view.php]: Ditto

File: view.php [hunk 14]
```diff
@@ -1,39 +1,50 @@
...
+	{
+		if($game[$key])
+		{
+			echo "$name: " . $game[$key] . "\n";
+		}
+	}
 ?>
-</pre>
+</pre>
```
Comment (puigru) [inline view.php]: Newline at EOF

File: view.php [hunk 15]
```diff
@@ -1,39 +1,50 @@
 <?php
+include 'common.php';
+
+$baseurl = 'http://infinity.unstable.life/Flashpoint/Data/Images';
+
+$keys =
+[
```
Comment (puigru) [inline view.php]: $keys = [

**Review by puigru — CHANGES_REQUESTED** [review]

File: index.php (hunk 6 above)
Comment (Seirade) [reply index.php:9]: It was just another external call/dependency, so it was mainly to cut down on that, but including it just for 3 icons seemed to be overkill.

File: index.php (hunk 7 above)
Comment (Seirade) [reply index.php]: If you want to add dependencies, I would suggest including a `package.json` and then running `npm install` so that they're always local.

Comment (puigru) [reply index.php]: Agreed, although it's out of scope for this PR.

File: view.php (hunk 12 above)
Comment (Seirade) [reply view.php]: They specifically throw the following errors in newer versions of PHP:
```
PHP Warning:  Use of undefined constant platform - assumed 'platform' (this will throw an Error in a future version of PHP) in FlashpointBrowser/search.php on line 36
PHP Warning:  Use of undefined constant title - assumed 'title' (this will throw an Error in a future version of PHP) in FlashpointBrowser/search.php on line 36
```
But I added it back in if that's the preference.

Comment (puigru) [reply view.php]: I'm referring to the if statement:
```php
if($game)
{
}
```
vs
```php
if ($game) {
}
```
What you're referring to is a separate issue in search.php

File: search.php (hunk 10 above)
Comment (Seirade) [reply search.php]: Someone's going to obsess over it not matching our `(18)` Extreme icon, and I personally don't like misusing symbols. I settled on just using a Bootstrap pill badge.

Comment (puigru) [reply search.php]: Sure

**Comment by Seirade** [comment]
Pushed changes, please review.

File: js/bootstrap.min.js [hunk 16]
```diff
@@ -0,0 +1,7 @@
+/*!
```
Comment (puigru) [inline js/bootstrap.min.js]: You forgot to remove these from the source tree

File: js/jquery-3.5.1.min.js [hunk 17]
```diff
@@ -0,0 +1,2 @@
+/*! jQuery v3.5.1 | (c) JS Foundation and other contributors | jquery.org/license */
```
Comment (puigru) [inline js/jquery-3.5.1.min.js]: This one too

File: view.php [hunk 18]
```diff
@@ -1,39 +1,46 @@
 <?php
+include 'common.php';
```
Comment (puigru) [inline view.php]: Missed this one earlier: require

File: view.php [hunk 19]
```diff
@@ -1,39 +1,46 @@
...
 $stmt = $db->prepare('SELECT * FROM game WHERE id = :id');
 $stmt->bindValue(':id', $id);
 $result = $stmt->execute();
 $game = $result->fetchArray(SQLITE3_ASSOC);
+$blur = isExtreme($game, $nsfw);
+
 if (!$game) {
-    header('HTTP/1.1 404 Not Found');
-    die('Game not found');
+	header('HTTP/1.1 404 Not Found');
```
Comment (puigru) [inline view.php]: Please replace tabs with spaces throughout this file

File: package.json [hunk 20]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "flashpointbrowser",
+  "version": "0.0.0",
```
Comment (puigru) [inline package.json]: Let's say 0.0.1?

File: .gitignore [hunk 21]
```diff
@@ -0,0 +1,4 @@
+node_modules
+package-lock.json
+preferences.json
+flashpoint.sqlite
```
Comment (puigru) [inline .gitignore]: Newline at EOF

**Review by puigru — CHANGES_REQUESTED** [review]
Almost there

File: common.php [hunk 22]
```diff
@@ -0,0 +1,25 @@
+<?php
+$json = json_decode(file_get_contents('preferences.json'), true);
+$nsfw = [];
+
+foreach($json['tagFilters'] as $filter)
```
Comment (puigru) [inline common.php]: Change brace style throughout this file

File: package.json (hunk 20 above)
Comment (Seirade) [reply package.json]: Sure, though I was just going to let you update that. Your branch before my changes would be 0.0.1, and after would be 0.0.2?

Comment (puigru) [reply package.json]: I don't really care as we've not been really tracking version so far. 0.0.1 works for me.

Comment (Seirade) [reply package.json]: Ok, I'll just do that then.
